jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ResponseTest.java

```diff
@@ -194,6 +196,80 @@ public boolean handle(Request request, Response response, Callback callback)
         assertThat(response.get(HttpHeader.LOCATION), is("/somewhere/else"));
     }
 
+    @Test
+    public void testXPoweredByDefault() throws Exception
```

Comment:
I'd like to see a test that calls `Response.reset()` and verifies those default headers are set again.

Reply:
The place where those headers are added is now in `HttpChannelState.onRequest`.
A call to `Response.reset()` erases those headers, and another `HttpChannelState.onRequest()` needs to occur.

This is true for `Server`, `X-Powered-By`, and `Date` headers.

Reply:
I've got the reset test in PR #10339